frontend/components/ifixit-image/ifixit-image.tsx

```diff
@@ -33,3 +59,26 @@ function isStrapiImage(src: string) {
       /^https:\/\/ifixit-(dev-)?strapi-uploads.s3.amazonaws.com\//
    );
 }
+
+function getImageLoader(sizeMap: SizeMap, defaultSize: string): ImageLoader {
+   return ({ src, width }: ImageLoaderProps) => {
+      const baseSrc = src.replace(/\.[^/.]+$/, '');
+      const sizeName = getImageSize(width, sizeMap, defaultSize);
+      // We don't use the ?width param server-side, but it gets rid of a nextjs warning
+      return baseSrc.concat('.', sizeName, `?width=${width}`);
+   };
+}
+
+function getImageSize(
+   width: number,
+   sizeMap: SizeMap,
+   defaultSize: string
+): string {
+   const sortedSizes = Object.entries(sizeMap).sort((a, b) => a[1] - b[1]);
```

Comment:
Might make sense to store the sizes as a sorted array of {width: int, name: string} rather than sorting the list every time.

Reply:
[2897234](https://github.com/iFixit/react-commerce/pull/537/commits/2897234e9f4dbc5f020c330885fc6e80ac06216a)